### PR TITLE
feat(unlock-app) - fix allow key granting buggy

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

fix key granting buggy 

https://github.com/unlock-protocol/unlock/assets/20865711/b297c004-86d5-47eb-a5b1-48151512333d

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx [hunk 1]
```diff
@@ -241,7 +248,7 @@ const ConnectStripe = ({
               variant="outlined-primary"
               className="w-full md:w-1/3"
               onClick={onGrantKeyRole}
-              loading={grantKeyGrantorRoleMutation.isLoading}
+              disabled={grantKeyGrantorRoleMutation.isLoading}
```
Comment (searchableguy) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx:251]: why the change?

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx [hunk 2]
```diff
@@ -151,7 +152,13 @@ const ConnectStripe = ({
   }
 
   const { isLoading: isLoadingCheckGrantedStatus, data: isGranted } = useQuery(
-    ['checkIsKeyGranter', lockAddress, network, keyGranter],
+    [
+      'checkIsKeyGranter',
+      lockAddress,
+      network,
+      keyGranter,
+      grantKeyGrantorRoleMutation.isSuccess,
```
Comment (searchableguy) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx:160]: why not use isEnabled here? 🤔

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx [hunk 3]
```diff
@@ -117,14 +117,15 @@ const ConnectStripe = ({
 
   const grantKeyGrantorRoleMutation = useMutation(async (): Promise<any> => {
     const walletService = await getWalletService(network)
-    return walletService.addKeyGranter({
+    return await walletService.addKeyGranter({
```
Comment (searchableguy) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx:120]: does await change behavior here? It shouldn't 🤔

Comment (kalidiagne) [reply unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx:120]: no, does not change it

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx (hunk 2 above)
Comment (kalidiagne) [reply unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx:160]: need to check `isGranted` status when i access the page or when `grantKeyGrantorRoleMutation` is done, 
with `enabled` will get stuck on page reload 🤔

Comment (searchableguy) [reply unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx]: No, that is fine. I'm saying you can use this is in enabled option right, `enabled: grantKeyGrantorRoleMutation.isSuccess`

Comment (searchableguy) [reply unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx]: This is going to be buggy. You should directly call refetch when mutation is called to get new data.

**Review by julien51 — APPROVED** [review]
